src/filter/mod.rs

```diff
@@ -77,23 +73,17 @@ pub trait Filter {
                 None => attempts += 1,
                 Some(event) => {
                     attempts = 0;
-                    match self.valve_state() {
-                        util::Valve::Open => match self.process(event, &mut events) {
-                            Ok(()) => for ev in events.drain(..) {
-                                util::send(&mut chans, ev)
-                            },
-                            Err(fe) => {
-                                error!(
-                                    "Failed to run filter with error: {:?}",
-                                    name_in_fe(&fe)
-                                );
-                                let event = event_in_fe(fe);
-                                util::send(&mut chans, event);
-                            }
+                    match self.process(event, &mut events) {
```

Comment:
From review:  We should decrement attempts here.

Reply:
I was wrong when we were discussing this. Line 75 has a reset to 0 in the success branch.